src/App.tsx

```diff
@@ -1,61 +1,123 @@
+import React, { useState } from 'react';
 import './App.scss';
+import { TodoList } from './components/TodoList';
+import { User } from './types/User';
+import { Todo } from './types/Todo';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+
+function getUser(userId: number): User | null {
+  const foundUser = usersFromServer.find(user => user.id === userId);
+
+  return foundUser || null;
+}
+
+const todos: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUser(todo.userId),
+}));
+
+const maxTodoid = todos
+  .map(todo => todo.id)
+  .reduce((a, b) => Math.max(a, b), 0) + 1;
 
 export const App = () => {
+  const [userId, setUserId] = useState(0);
+  const [isUserChose, setIsUserChose] = useState(true);
+  const [title, setTitle] = useState('');
+  const [isTitleChose, setIsTitleChose] = useState(true);
+
+  const handleTitleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTitle(event.target.value);
+    setIsTitleChose(true);
+  };
+
+  const handleUserChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setUserId(Number(event.target.value));
+    setIsUserChose(true);
+  };
+
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    if (!userId) {
+      setIsUserChose(false);
+    }
+
+    if (!title) {
+      setIsTitleChose(false);
+    }
+
+    if (!userId || !title) {
+      return;
+    }
```

Comment:
I can create todo where the title has just spaces
<img width="170" alt="Screenshot 2022-09-26 at 22 21 21" src="https://user-images.githubusercontent.com/47606379/192362033-a5753798-256b-4394-99da-51df0f1e16f7.png">